File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        n = len(colors)
        i = 0
        total_cost = 0
        while i < n:
            mx = 0
            sm = 0
            for j in range(i, n):
                if colors[j] != colors[i]:
                    i = j
                    break
                sm += neededTime[j]
                if neededTime[j] > mx:
                    mx = neededTime[j]
            else:
                i = n
            total_cost += sm - mx
        return total_cost

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        int n = neededTime.length;
        int i = 0;
        int totalTime = 0;
        
        while (i < n) {
            int mx = 0;
            int sm = 0;
            int j = i;
            for (; j < n; j++) {
                if (colors.charAt(j) != colors.charAt(i)) {
                    i = j;
                    break;
                }
                
                sm += neededTime[j];
                if (neededTime[j] > mx) {
                    mx = neededTime[j];
                }
            }
            
            totalTime += sm - mx;
            
            if (j == n) {
                break;
            }
        }
        
        return totalTime;
    }
}